Assets/HoloToolkit/Boundary/Scripts/BoundaryManager.cs

```diff
@@ -0,0 +1,162 @@
+
+#if UNITY_EDITOR
+                // So the floor quad does not occlude in editor testing, draw it lower.
+                floorQuadInstance.transform.localPosition = new Vector3(0, -3, 0);
+#else
+                // Inside immersive headset draw floor quad at Y value of dimensions.
+                Vector3 dimensions;
+                // TODO: BUG: Unity: configured bool always returns false.
+                // TODO: BUG: Unity: TryGetDimensions does not return true either.
+                //if (UnityEngine.Experimental.XR.Boundary.configured &&
+                //    UnityEngine.Experimental.XR.Boundary.TryGetDimensions(out dimensions))
+
+                if (UnityEngine.Experimental.XR.Boundary.TryGetDimensions(out dimensions))
+                {
+                    if (dimensions != null)
+                    {
+                        Debug.Log("Drawing floor at dimensions Y.");
+                        // Draw the floor at boundry Y.                    
+                        floorQuadInstance.transform.localPosition = new Vector3(0, dimensions.y, 0);
+                    }
+                }
+                else
+                {
+                    Debug.Log("Drawing floor at 0,0,0.");
+                    // Draw the floor at 0,0,0.
+                    floorQuadInstance.transform.localPosition = Vector3.zero;
+                }
+#endif
+                floorQuadInstance.SetActive(true);
+            }
+        }
+
+        /// <summary>
+        /// Pass in the game object's position to check if it's within 
+        /// the specified boundary space.
+        /// </summary>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        public bool ContainsObject(Vector3 gameObjectPosition)
+        {
+            // TODO: BUG: Unity
+            //if (!UnityEngine.Experimental.XR.Boundary.configured)
+            //{
+            //    Debug.Log("Boundary not configured.");
+            //    return;
+            //}
+
+            // Get all the bounds setup by the user.
+            List<Vector3> boundaryGeometry = new List<Vector3>();
+            if (UnityEngine.Experimental.XR.Boundary.TryGetGeometry(boundaryGeometry))
+            {
+                if (boundaryGeometry != null)
+                {
+                    // Create a UnityEngine.Bounds volume with those values.
+                    foreach (Vector3 boundaryGeo in boundaryGeometry)
+                    {
+                        boundaryBounds.Encapsulate(boundaryGeo);
+                    }
+                }
+            }
+
+            // Ensuring that we set height of the bounds volume to be say 10 feet tall.
+            boundaryBounds.Encapsulate(new Vector3(0, boundaryHeight, 0));
+
+            // Check if the supplied game object's position is within the bounds volume.
+            if (boundaryBounds != null && HolographicSettings.IsDisplayOpaque)
```

Comment:
Why do we check IsDisplayOpaque here?

Reply:
That's true, but I wasn't sure if it'd be better to check if the bounds are empty/not set-up instead, so we aren't locking ourselves into a specific display type vs bounds actually being supported/present.